### DISCUSSION TITLE
TRIQS Scaling or Benchmarking on HPC Cluster

### DISCUSSION THREAD

**Discussion by hrishitbanerjee** [TRIQS/triqs#828] [General]
Hi,
Is there any scaling or benchmarks for TRIQS/CTHYB/DFTTools on HPC Clusters available.
Kindly help.
Thanks and best regards
Hrishit

**Comment by the-hampel** [comment]
Hi Hrishit, 
I just converted this into a discussion, because I think this is not really an issue. 

I have some benchmark data for solid_dmft (https://flatironinstitute.github.io/solid_dmft) which uses dfttools, triqs, and cthyb for a one-shot dmft calculation for LuNiO3 (two impurity problems). I would say that cthyb scales perfectly, so all deviations from the optimal linear scaling are from the k-sums and small things need to be done in postprocessing etc in dfttools. This data is now 1-2 years old, but it should be still more or less valid. Especially compared to DFT codes this scaling is near perfect, but I believe that one could optimize the scaling even further. I attached the plots. The first one is the scaling, and the second one the efficiency, i.e. the deviation from perfect scaling.
![benchmark_solid_dmft](https://user-images.githubusercontent.com/21337117/142004677-cef09c00-4c81-4f1d-abed-fcf866757aab.png)
![benchmark_solid_dmft_eff](https://user-images.githubusercontent.com/21337117/142004684-365a0db9-5ddb-4110-94f5-bc5bdc6060e4.png)

**Comment by hrishitbanerjee** [comment]
Hi
Thanks for the plots. This is also very helpful.

**Comment by hrishitbanerjee** [comment]
I did some calculations myself on NiO, 1 impurity, 1 iteration of DMFT, with a total of 2000000 simulation steps after 200000 warm up steps on different number of processors and this is what I obtained.
![triqs-benchmark-scale](https://user-images.githubusercontent.com/40570674/142018247-cb72ba43-288a-4bef-9b48-e661b1ab3b71.png)
